dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreCollectionSqlBuilder.cs

```diff
@@ -0,0 +1,447 @@
+
+        StringBuilder createTableCommand = new();
+        createTableCommand.AppendLine($"CREATE TABLE {(ifNotExists ? "IF NOT EXISTS " : "")}{schema}.\"{tableName}\" (");
+
+        // Add the key column
+        var keyPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPostgresTypeName(keyProperty.PropertyType);
+        createTableCommand.AppendLine($"    \"{keyName}\" {keyPgTypeInfo.PgType} {(keyPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+
+        // Add the data columns
+        foreach (var dataProperty in dataProperties)
+        {
+            string columnName = dataProperty.StoragePropertyName ?? dataProperty.DataModelPropertyName;
+            var dataPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPostgresTypeName(dataProperty.PropertyType);
+            createTableCommand.AppendLine($"    \"{columnName}\" {dataPgTypeInfo.PgType} {(dataPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+        }
+
+        // Add the vector columns
+        foreach (var vectorProperty in vectorProperties)
+        {
+            string columnName = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+            var vectorPgTypeInfo = PostgresVectorStoreRecordPropertyMapping.GetPgVectorTypeName(vectorProperty);
+            createTableCommand.AppendLine($"    \"{columnName}\" {vectorPgTypeInfo.PgType} {(vectorPgTypeInfo.IsNullable ? "" : "NOT NULL")},");
+        }
+
+        createTableCommand.AppendLine($"    PRIMARY KEY (\"{keyName}\")");
+
+        createTableCommand.AppendLine(");");
+
+        return new PostgresSqlCommandInfo(commandText: createTableCommand.ToString());
+    }
+
+    /// <inheritdoc />
+    public PostgresSqlCommandInfo BuildCreateVectorIndexCommand(string schema, string tableName, VectorStoreRecordVectorProperty vectorProperty)
+    {
+        var vectorColumnName = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+        // Only support creating HNSW index creation through the connector.
+        var indexTypeName = vectorProperty.IndexKind switch
+        {
+            IndexKind.Hnsw => "hnsw",
+            _ => throw new NotSupportedException($"Index kind '{vectorProperty.IndexKind}' is not supported for table creation. If you need to create an index of this type, please do so manually. Only HNSW indexes are supported through the vector store.")
+        };
+
+        var indexOps = vectorProperty.DistanceFunction switch
+        {
+            DistanceFunction.CosineDistance => "vector_cosine_ops",
+            DistanceFunction.CosineSimilarity => "vector_cosine_ops",
+            DistanceFunction.DotProductSimilarity => "vector_ip_ops",
+            DistanceFunction.EuclideanDistance => "vector_l2_ops",
+            DistanceFunction.ManhattanDistance => "vector_l1_ops",
+            null => throw new ArgumentException("Distance function must be specified for HNSW index."),
```

Comment:
We should just default to cosine if not specified.

Reply:
Will change to defaulting cosine here, and also for the query side, which is currently defaulting to L2